test/junit/src/test/scala/strawman/collection/immutable/LazyListTest.scala

```diff
@@ -139,4 +140,28 @@ class LazyListTest {
     assertEquals(LazyList(3), LazyList(1, 2, 3).drop(2))
     assertEquals(LazyList(3, 4), LazyList(1, 2, 3, 4).drop(2))
   }
+
+  @Test
+  def testForceReturnsEvaluatedLazyList() : Unit = {
+    var i = 0
+    def f: Int = { i += 1; i }
+    val xs = LazyList.from(Iterator.fill(3)(f))
+    assertEquals(0, i)
+    xs.force
+    assertEquals(3, i)
+    // it's possible to implement `force` with incorrect string representation
+    // (to forget about `tlEvaluated` update)
+    assertEquals( "1 #:: 2 #:: 3 #:: Empty", xs.toString())
+  }
+
+  @Test
+  def testSameElements(): Unit = {
+    assert(LazyList().sameElements(LazyList()))
+    assert(!LazyList().sameElements(LazyList(1)))
+    assert(LazyList(1,2).sameElements(LazyList(1,2)))
+    assert(!LazyList(1,2).sameElements(LazyList(1)))
+    assert(!LazyList(1).sameElements(LazyList(1,2)))
+    assert(!LazyList(1).sameElements(LazyList(2)))
+  }
+
```

Comment:
Can you also add a test with a cyclic collection?

Reply:
I added 2 tests. 
At the same time comparison of equal collections but with different references is infinite. I didn't catch OOM. Strange feeling but it seems ok - collection doesn't have unnecessary overhead and indeed infinite. But should I add test for absent of OOM? And if so how is it possible to implement?
```
val cycle1: LazyList[Int] = 1 #:: 2 #:: cycle1
val cycle2: LazyList[Int] = 1 #:: 2 #:: 3 #:: cycle2
val cycle3: LazyList[Int] = cycle2 #::: cycle3
val cycle4: LazyList[Int] = cycle2 #::: cycle4
assert(!cycle3.sameElements(cycle4))
```

One more thing (applies for old collections too). `val cycle: LazyList[Int] = LazyList.Empty #::: cycle` evaluates into 
```
java.lang.NullPointerException
  at strawman.collection.immutable.LazyList$.from(LazyList.scala:523)
  at strawman.collection.immutable.LazyList$.from(LazyList.scala:444)
  at strawman.collection.immutable.LazyListOps.lazyAppendAll(LazyList.scala:257)
  at strawman.collection.immutable.LazyListOps.lazyAppendAll$(LazyList.scala:256)
  at strawman.collection.immutable.LazyList.lazyAppendAll(LazyList.scala:186)
  at strawman.collection.immutable.LazyList$Deferrer.$hash$colon$colon$colon(LazyList.scala:513)
  ... 40 elided
```
Didn't have enough time to investigate. Is it ok (look back to the old collection) or should be fixed?

Reply:
let's create a ticket to keep track of it